### PR TITLE
:bug: code cleaning and hotfix for #105

### PR DESCRIPTION
**Pull request by pad92**
add `LOG_FILE = '/app/alertad.log'` into alertad.conf with entrypoint

### CODE REVIEW

File: slash/app/uwsgi.ini [hunk 1]
```diff
@@ -11,7 +11,7 @@ socket = /tmp/uwsgi.sock
 buffer-size = 8192
 chmod-socket = 664
 uid = alerta
-gid = alerta
+gid = root
```
Comment (pad92) [inline slash/app/uwsgi.ini]: alerta created with gid 0

File: slash/app/uwsgi.ini [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [uwsgi]
-chdir = /
+chdir = /app
```
Comment (pad92) [inline slash/app/uwsgi.ini]: with / ? :)

File: slash/docker-entrypoint.sh [hunk 3]
```diff
@@ -12,15 +12,16 @@ fi
 if [ ! -f "${ALERTA_SVR_CONF_FILE}" ]; then
   cat >"${ALERTA_SVR_CONF_FILE}" << EOF
 SECRET_KEY = '$(< /dev/urandom tr -dc A-Za-z0-9_\!\@\#\$\%\^\&\*\(\)-+= | head -c 32)'
+LOG_FILE = '/app/alertad.log'
```
Comment (pad92) [inline slash/docker-entrypoint.sh]: my hotfix

File: Dockerfile [hunk 4]
```diff
@@ -7,62 +7,57 @@ ARG BUILD_DATE
...
+    && /venv/bin/pip install uwsgi alerta alerta-server==${VERSION}
+
+ENV PATH ${PATH}:/venv/bin
+
+ADD https://github.com/alerta/angular-alerta-webui/archive/v${VERSION}.tar.gz /tmp/angular-alerta-webui.tar.gz
+RUN tar zxf /tmp/angular-alerta-webui.tar.gz -C /tmp \
+    && mv /tmp/angular-alerta-webui-${VERSION}/app /web \
+    && mv /web/config.json /web/config.json.orig \
+    && rm -fr /tmp/angular-alerta-webui.tar.gz \
+              /tmp/angular-alerta-webui-${VERSION}
```
Comment (pad92) [inline Dockerfile]: remove tmp files

File: Dockerfile [hunk 5]
```diff
@@ -7,62 +7,57 @@ ARG BUILD_DATE
...
-COPY wsgi.py /app/wsgi.py
-COPY uwsgi.ini /app/uwsgi.ini
-COPY nginx.conf /app/nginx.conf
-
-RUN ln -sf /dev/stdout /var/log/nginx/access.log
-RUN ln -sf /dev/stdout /var/log/nginx/error.log
-RUN chgrp -R 0 /app /venv /web && \
-    chmod -R g=u /app /venv /web && \
-    useradd -u 1001 -g 0 alerta
+RUN apt-get update \
```
Comment (DisasteR) [inline Dockerfile]: Remove this line

File: Dockerfile [hunk 6]
```diff
@@ -7,62 +7,57 @@ ARG BUILD_DATE
...
+    && DEBIAN_FRONTEND=noninteractive apt-get upgrade -y --no-install-recommends \
```
Comment (DisasteR) [inline Dockerfile]: Remove this line

File: Dockerfile [hunk 7]
```diff
@@ -7,62 +7,57 @@ ARG BUILD_DATE
...
+      libffi-dev \
+      libldap2-dev \
+      libpq-dev \
+      libsasl2-dev \
+      mongodb-clients \
+      nginx-light \
+      postgresql-client \
+      python3-dev \
+      supervisor \
+      wget
```
Comment (DisasteR) [inline Dockerfile]: Add Apt cleanup

File: Dockerfile [hunk 8]
```diff
@@ -7,62 +7,57 @@ ARG BUILD_DATE
...
+
+COPY slash/ /
+
+RUN chgrp -R 0 /app /venv /web \
+    && chmod -R g=u /app /venv /web \
+    && useradd -r -d /app -u 1001 -g 0 alerta
 
 USER 1001
 
 ENV ALERTA_SVR_CONF_FILE /app/alertad.conf
```
Comment (DisasteR) [inline Dockerfile]: Merge all env following layers

**Comment by satterly** [comment]
Thanks for your contribution however the "fix" is to not write local log files to the docker container in the first place. This was a misconfiguration of the new logging that was introduced in the latest version and I'm pushing out a fix now.

**Comment by pad92** [comment]
Ok, I've cleaned up, if you want to merge the optimizations

**Comment by satterly** [comment]
Thanks! I'll take a look when I get a chance. 👍

**Pull request closed without merging**